product_multi_barcode_constraint_per_company/models/product_barcode.py

```diff
@@ -0,0 +1,35 @@
+# Copyright 2023 Cetmix OÜ
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import _, api, models
+from odoo.exceptions import UserError
+
+
+class ProductBarcode(models.Model):
+    _inherit = "product.barcode"
+
+    @api.constrains("name")
+    def _check_duplicates(self):
+        barcodes = self.search(
+            [("id", "not in", self.ids), ("name", "in", self.mapped("name"))]
+        )
+
+        for barcode in barcodes:
+            for record in self:
```

Comment:
Cycle into cycle it's bad solution, please leave only one cycle.

Reply:
okay, I'll fix it